napari/layers/volume/volume.py

```diff
@@ -0,0 +1,245 @@
+from vispy.scene.visuals import Volume as VolumeNode
+from vispy.visuals.transforms import STTransform
+from ..base import Layer
+from ...util.misc import calc_data_range, increment_unnamed_colormap
+from ...util.event import Event
+from ...util.colormaps import COLORMAPS_3D_DATA
+from ._constants import Rendering
+
+
+class Volume(Layer):
+    """Volume layer.
+
+    Parameters
+    ----------
+    volume : array
+        Volume data. Is 3 dimensional. If the last dimension (channel) is
+        3 or 4 can be interpreted as RGB or RGBA if multichannel is `True`.
+    metadata : dict, keyword-only
+        Volume metadata.
+    colormap : str, vispy.Color.Colormap, tuple, dict, keyword-only
+        Colormap to use for luminance volumes. If a string must be the name
+        of a supported colormap from vispy or matplotlib. If a tuple the
+        first value must be a string to assign as a name to a colormap and
+        the second item must be a Colormap. If a dict the key must be a
+        string to assign as a name to a colormap and the value must be a
+        Colormap.
+    clim : list (2,), keyword-only
+        Color limits to be used for determining the colormap bounds for
+        luminance volumes. If not passed is calculated as the min and max of
+        the volume.
+    clim_range : list (2,), keyword-only
+        Range for the color limits. If not passed is be calculated as the
+        min and max of the volume. Passing a value prevents this calculation
+        which can be useful when working with very large datasets that are
+        dynamically loaded.
+    name : str, keyword-only
+        Name of the layer.
+
+    Attributes
+    ----------
+    data : array
+        Volume data. Should be 3 dimensional
+    metadata : dict
+        Volume metadata.
+    colormap : 2-tuple of str, vispy.color.Colormap
+        The first is the name of the current colormap, and the second value is
+        the colormap. Colormaps are used for luminance volumes.
+    colormaps : tuple of str
+        Names of the available colormaps.
+    clim : list (2,) of float
+        Color limits to be used for determining the colormap bounds for
+        luminance volumes.
+    clim_range : list (2,) of float
+        Range for the color limits for luminace volumes.
+
+    Extended Summary
+    ----------
+    _data_view : array (N, M, K)
+        Volume data for the currently viewed slice.
+    """
+
+    class_keymap = {}
+    _colormaps = COLORMAPS_3D_DATA
+    _default_rendering = Rendering.MIP.value
+    _translate = (0, 0, 0)
+    _scale = (0.09, 0.09, 0.09, 1)
+
+    def __init__(
+        self,
+        volume,
+        *,
+        metadata=None,
+        colormap='fire',
+        clim=None,
+        clim_range=None,
+        name=None,
+        **kwargs,
+    ):
+
+        visual = VolumeNode(volume, threshold=0.225, emulate_texture=False)
+        print(self._translate)
+        print(self._scale)
+        visual.transform = STTransform(
```

Comment:
why are you setting the `visual.transform` here? this is done automatically in the `_visual_wrapper.py`. can you delete this, the print statements, and the `_scale` and `_translate` properties on this layer. If the camera position is not right we should fix it in the QtViewer rather than here

Reply:
I think it is the scale with which the volume is displayed and is linked to the volume node how it should be scaled not the camera. I am removing the print statements but the problem still exists in the display

Reply:
STTransform (we are actually setting STTransform in VisualWrapper class that is inherited by layers) by default doesn't do any scaling or transformation - https://github.com/vispy/vispy/blob/master/vispy/visuals/transforms/linear.py#L82  as our 2D images don't need that. But we have to find an optimal value or a formula for optimal value from the shape of 3D volume so it all fits inside the window.  hence the numbers 50, 50, 1 etc inside the volume layer